### PR TITLE
sound encoder and demo

### PR DESCRIPTION
**Pull request by tomsilver**
@scottpurdy please review

cc @rhyolight

### CODE REVIEW

File: sound_encoder/sound_encoder.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
```
Comment (scottpurdy) [inline sound_encoder/sound_encoder.py]: year

File: sound_encoder/live_sound_encoding_demo.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2013, Numenta, Inc.  Unless you have an agreement
```
Comment (scottpurdy) [inline sound_encoder/live_sound_encoding_demo.py]: year

File: sound_encoder/live_sound_encoding_demo.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+CHANNELS = 2
+RATE = 44100
+RECORD_SECONDS = 10
+
+
+
+def getAudioStream():
+  p = pyaudio.PyAudio()
+  return p.open(format=FORMAT,
+				channels=CHANNELS,
```
Comment (scottpurdy) [inline sound_encoder/live_sound_encoding_demo.py]: Looks like there are tabs here. Pls use 2 space indents

**Comment by scottpurdy** [comment]
Please run pylint on the files and removes tabs, etc. and I will review after that

**Comment by tomsilver** [comment]
@scottpurdy thanks for feedback, incorporated and ready for re-review

**Comment by scottpurdy** [comment]
live_sound_encoding_demo.py still has tabs in it

File: sound_encoder/live_sound_encoding_demo.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  n = 300
+  w = 31
+  minval = 20
+  maxval = 10000
+
+  soundEncoder = SoundEncoder(n, w, RATE, CHUNK, minval, maxval)
+  stream = getAudioStream()
+  sdrs = recordAndEncode(stream, soundEncoder)
+  visualizeSDRs(sdrs)
+
```
Comment (scottpurdy) [inline sound_encoder/live_sound_encoding_demo.py]: remove blank line

File: sound_encoder/sound_encoder.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    self.w = w
+    self.rate = rate
+    self.chunk  = chunk
+    self.minval = minval
+    self.maxval = maxval
+    self.name = name
+    self._scalarEncoder = ScalarEncoder(name="scalar_"+str(name), n=n, w=w,
+                                        minval=minval, maxval=maxval)
+
+  def _detectFrequency(self, inputArr):
```
Comment (scottpurdy) [inline sound_encoder/sound_encoder.py]: two blank lines before methods (elsewhere as well)

File: sound_encoder/sound_encoder.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    """Use FFT to find maximum frequency present in the input."""
+    fftData=abs(np.fft.rfft(inputArr))**2
+    maxFreqIdx = np.argmax(fftData)
+
+    if maxFreqIdx < len(fftData)-1:
+      # Quadratic interpolation
+      y0,y1,y2 = np.log(fftData[maxFreqIdx-1:maxFreqIdx+2:])
```
Comment (scottpurdy) [inline sound_encoder/sound_encoder.py]: put a space after each comma

**Comment by tomsilver** [comment]
@scottpurdy thanks, incorporated feedback

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**